plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,701 @@
+      - If not specified, a list of all objects in the specified bucket will be returned.
+    required: false
+    type: str
+  object_details:
+    description:
+      - Retrieve requested S3 object detailed information.
+    required: false
+    type: dict
+    suboptions:
+      object_acl:
+        description:
+          - Retreive S3 object ACL.
+        required: false
+        type: bool
+        default: false
+      object_legal_hold:
+        description:
+          - Retreive S3 object legal_hold.
+        required: false
+        type: bool
+        default: false
+      object_lock_configuration:
+        description:
+          - Retreive S3 object lock_configuration.
+        required: false
+        type: bool
+        default: false
+      object_retention:
+        description:
+          - Retreive S3 object retention.
+        required: false
+        type: bool
+        default: false
+      object_tagging:
+        description:
+          - Retreive S3 object Tags.
+        required: false
+        type: bool
+        default: false
+      object_attributes:
+        description:
+          - Retreive S3 object attributes.
+        required: false
+        type: bool
+        default: false
+      attributes_list:
+        description:
+          - The fields/details that should be returned.
+          - Required when I(object_attributes) is C(true) in I(object_details).
+        type: list
+        elements: str
+        choices: ['ETag', 'Checksum', 'ObjectParts', 'StorageClass', 'ObjectSize']
+
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+
+'''
+
+EXAMPLES = r'''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Retrieve a list of objects in S3 bucket
+  amazon.aws.s3_object_info:
+    bucket_name: MyTestBucket
+
+- name: Retrieve object metadata without object itself
+  amazon.aws.s3_object_info:
+    bucket_name: MyTestBucket
+    object_name: MyTestObjectKey
+
+- name: Retrieve detailed S3 information for all objects in the bucket
+  amazon.aws.s3_object_info:
+    bucket_name: MyTestBucket
+    object_details:
+      object_acl: true
+      object_attributes: true
+      attributes_list:
+        - ETag
+        - ObjectSize
+        - StorageClass
+
+- name: Retrieve detailed S3 object information
+  amazon.aws.s3_object_info:
+    bucket_name: MyTestBucket
+    object_name: MyTestObjectKey
+    object_details:
+      object_acl: true
+      object_tagging: true
+      object_legal_hold: true
+      object_attributes: true
+      attributes_list:
+        - ETag
+        - ObjectSize
+
+'''
+
+RETURN = r'''
+s3_keys:
+  description: List of object keys.
+  returned: when only I(bucket_name) is specified and I(object_name), I(object_details) are not specified.
+  type: list
+  elements: str
+  sample:
+  - prefix1/
+  - prefix1/key1
+  - prefix1/key2
+object_info:
+    description: S3 object details.
+    returned: always
```

Comment:
Are you sure `object_info` is always returned?

Reply:
Fixed.